### PR TITLE
[#108127818] Fixed issues with token id that does not match incoming account id

### PR DESCRIPTION
**Pull request by hovsepm**

### CODE REVIEW

**Comment by hovsepm** [comment]
on-demand http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/354/

File: src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs [hunk 1]
```diff
@@ -90,7 +91,26 @@ public RMProfileClient(AzureRMProfile profile)
...
                     {
                         token = AcquireAccessToken(account, environment, tenant, password, ShowDialog.Auto);
-                        account.SetOrAppendProperty(AzureAccount.Property.Tenants, tenant);
+
+                        if (accountId == null)
+                        {
+                            accountId = account.Id;
+                            account.SetOrAppendProperty(AzureAccount.Property.Tenants, tenant);
+                        }
+                        else if (accountId == account.Id)
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs]: We probably want to control the strong comparison, I would use string.Equals, with StringComparison.OrdinalIgnoreCase

File: src/ResourceManager/Profile/Commands.Profile/Properties/Resources.resx [hunk 2]
```diff
@@ -120,6 +120,9 @@
   <data name="AccessTokenRequiresAccount" xml:space="preserve">
     <value>AccountId must be provided to use an AccessToken credential.</value>
   </data>
+  <data name="AccountIdMismatch" xml:space="preserve">
+    <value>AccountId '{0}' for tenant '{1}' does not match home accountId '{2}'</value>
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Properties/Resources.resx]: <nit> Account ID</nit>

**Comment by hovsepm** [comment]
new on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/360/

**Pull request merged**